### DISCUSSION TITLE
Numerical instability issue as MPI size

### DISCUSSION THREAD

**Discussion by kimu-5834** [firemodels/fds#12628] [Q&A]
Dear developers,

I tried to test my FDS case on a cluster computer in a large MPI environment. The problem is that no numerical instability issue occurs with a small MPI setting (20), but numerical instability issue arises when I increase the setting to around 240 MPI, it shows numerical instability as below. I have maintained the same mesh size for both cases. I would appreciate it if you could let me know the reason.

 Starting FDS ...

 MPI Process      0 started on b01-6202c

 Reading FDS input file ...

 Fire Dynamics Simulator

 Current Date     : March 13, 2024  11:52:17
 Revision         : -
 Revision Date    : 
 Compilation Date :  2月 28, 2023  17:18:39

 MPI Enabled;    Number of MPI Processes:     240
 OpenMP Enabled; Number of OpenMP Threads:      2

 MPI version: 3.1
 MPI library version: FUJITSU MPI Library 4.0.0 (4.0.1fj4.0.0)

 Job TITLE        : 
 Job ID string    : base_mesh5_upper10cm_8_255_240_new

 Time Step:      1, Simulation Time:      0.04 s
 Time Step:      2, Simulation Time:      0.07 s
 Time Step:      3, Simulation Time:      0.07 s
 Time Step:      4, Simulation Time:      0.07 s
 Time Step:      5, Simulation Time:      0.07 s
 Time Step:      6, Simulation Time:      0.08 s
 Time Step:      7, Simulation Time:      0.08 s
 Time Step:      8, Simulation Time:      0.08 s
 Time Step:      9, Simulation Time:      0.08 s
 Time Step:     10, Simulation Time:      0.08 s
 Time Step:     20, Simulation Time:      0.09 s
 Time Step:     30, Simulation Time:      0.10 s
 Time Step:     40, Simulation Time:      0.12 s
 Time Step:     50, Simulation Time:      0.13 s
 Time Step:     60, Simulation Time:      0.13 s
 Time Step:     70, Simulation Time:      0.14 s
 Time Step:     80, Simulation Time:      0.15 s
 Time Step:     90, Simulation Time:      0.18 s
 Time Step:    100, Simulation Time:      0.18 s
 Time Step:    105, Simulation Time:      0.18 s

ERROR: Numerical Instability - FDS stopped (CHID: base_mesh5_upper10cm_8_255_240_new)

below is my FDS input file for 20MPI and 240MPI.

[Test_case_for_20_240_mpi.zip](https://github.com/firemodels/fds/files/14609935/Test_case_for_20_240_mpi.zip)

**Comment by Er9y714** [comment]
Both have the same domain but one has 20 meshes, another 240. In every mesh boundary an error is introduced. Therefore you should tighten (set smaller than default) the velocity and pressure tolerances. See user guide VELOCITY_TOLERANCE and PRESSURE_TOLERANCE. You might also increase the max number of pressure iterations.

**Comment by mcgratta** [comment]
```
Fire Dynamics Simulator

Current Date : March 13, 2024 11:52:17
Revision : -
Revision Date :
Compilation Date : 2月 28, 2023 17:18:39
```
It appears that you compiled the code yourself, but in Two-moon 2023. Try compiling the latest source code in the FDS repository.

**Comment by mcgratta** [comment]
The case is running for me
```
 Fire Dynamics Simulator

 Current Date     : March 15, 2024  09:23:20
 Revision         : FDS-6.8.0-1682-gfa4aa15-master
 Revision Date    : Mon Mar 11 22:14:33 2024 -0400
 Compiler         : Intel(R) Fortran Intel(R) 64 Compiler Classic for applications running on Intel(R) 64, Version 2021.7.1 Build 20221019_000000
 Compilation Date : Mar 12, 2024 11:14:41

 Number of MPI Processes:  240

 MPI version: 3.1
 MPI library version: Intel(R) MPI Library 2021.7 for Linux* OS


 Job TITLE        :
 Job ID string    : base_mesh5

 Time Step:      1, Simulation Time:      0.04 s
 Time Step:      2, Simulation Time:      0.07 s
 Time Step:      3, Simulation Time:      0.07 s
 Time Step:      4, Simulation Time:      0.08 s
 Time Step:      5, Simulation Time:      0.08 s
...
```
